aten/src/ATen/native/cuda/ConvolutionMM2d.cu

```diff
@@ -0,0 +1,67 @@
+#include <ATen/ATen.h>
+
+namespace at {
+namespace native {
+
+namespace {
+
+std::tuple<Tensor&, Tensor&, Tensor&> slow_conv2d_backward_out_cuda(
+    Tensor& grad_input,
+    Tensor& grad_weight,
+    Tensor& grad_bias,
+    const Tensor& grad_output,
+    const Tensor& self,
+    const Tensor& weight,
+    IntArrayRef kernel_size,
+    IntArrayRef stride,
+    IntArrayRef padding,
+    const Tensor& finput,
+    const Tensor& fgrad_input) {
+  if (grad_weight.defined()) {
+    grad_weight.resize_(weight.sizes());
+    grad_weight.zero_();
+  }
+  if (grad_bias.defined()) {
+    grad_bias.resize_({ weight.size(0) });
+    grad_bias.zero_();
+  }
+  return legacy::cuda::_thnn_conv2d_backward_out(grad_input, grad_weight, grad_bias,
+                                                 grad_output, self, weight,
+                                                 kernel_size, stride, padding,
+                                                 finput, fgrad_input);
+}
+
+std::tuple<Tensor, Tensor, Tensor> slow_conv2d_backward_cuda(
+    const Tensor& grad_output,
+    const Tensor& self,
+    const Tensor& weight,
+    IntArrayRef kernel_size,
+    IntArrayRef stride,
+    IntArrayRef padding,
+    const Tensor& finput,
+    const Tensor& fgrad_input,
+    std::array<bool, 3> output_mask) {
+  Tensor grad_input;
+  Tensor grad_weight;
+  Tensor grad_bias;
+
+  if (output_mask[0]) {
+    grad_input = at::empty({0}, grad_output.options());
+  }
+
+  if (output_mask[1]) {
+    grad_weight = at::empty({0}, grad_output.options());
+  }
+
+  if (output_mask[2]) {
+    grad_bias = at::empty({0}, grad_output.options());
+  }
+
+  return native::slow_conv2d_backward_out_cuda(grad_input, grad_weight, grad_bias,
+                                               grad_output, self, weight,
+                                               kernel_size, stride, padding,
+                                               finput, fgrad_input);
+}
```

Comment:
Are these just a straight transcription of the generated code or is there more to think about here

Reply:
it's not quite a straight transcription, but it is the same as what was already done on the CPU side.  The two differences are:
- the generated non-out variant doesn't call the out variant.  This allows codegen to do some stuff specifically for the out variant but isn't necessary in this case.
- The creation of the outputs in the non-out variants are done in a way that avoids dispatch but is so verbose that no one would actually write it:
```
    auto grad_bias_ = output_mask[2] ? c10::make_intrusive<TensorImpl, UndefinedTensorImpl>(c10::Storage(scalarTypeToTypeMeta(dispatch_scalar_type), 0, allocator(), true),DispatchKey::CUDA).releas\
e() : nullptr;
    auto grad_bias = Tensor(c10::intrusive_ptr<TensorImpl, UndefinedTensorImpl>::reclaim(grad_bias_ == nullptr ? (TensorImpl*)UndefinedTensorImpl::singleton() : (TensorImpl*)grad_bias_));
```